src/main/java/ch/cern/eam/wshub/core/services/workorders/impl/MECServiceImpl.java

```diff
@@ -0,0 +1,166 @@
+     * @throws InforException
+     */
+    @Override
+    public String addWorkOrderEquipment(InforContext context, String workOrderID, MEC mecProperties) throws InforException {
+        MECService.validateInput(workOrderID, mecProperties);
+        
+        mecProperties.setWorkorderID(workOrderID);
+
+        MP7394_AddWorkOrderEquipment_001 mp7394_addWorkOrderEquipment_001 = new MP7394_AddWorkOrderEquipment_001();
+        net.datastream.schemas.mp_entities.workorderequipment_001.WorkOrderEquipment workOrderEquipment = new net.datastream.schemas.mp_entities.workorderequipment_001.WorkOrderEquipment();
+
+        tools.getInforFieldTools().transformWSHubObject(workOrderEquipment, mecProperties, context);
+        mp7394_addWorkOrderEquipment_001.getWorkOrderEquipment().add(workOrderEquipment);
+        MP7394_AddWorkOrderEquipment_001_Result res = tools.performInforOperation(context, inforws::addWorkOrderEquipmentOp, mp7394_addWorkOrderEquipment_001);
+
+        return res.getResultData().getRELATEDWORKORDERID().get(0).getJOBNUM();
+    }
+
+    /**
+     * Deletes an equipment from the specified parent workorder.
+     *
+     * @param context           the user credentials
+     * @param parentWorkorderID the ID of the parent workorder
+     * @param equipmentID       the ID of the equipment to delete
+     * @return
+     * @throws InforException
+     */
+    @Override
+    public String deleteWorkOrderEquipment(InforContext context, String parentWorkorderID, String equipmentID) throws InforException {
+        MECService.validateInput(parentWorkorderID, equipmentID);
+
+        MP7396_RemoveWorkOrderEquipment_001 mp7396_removeWorkOrderEquipment_001 = new MP7396_RemoveWorkOrderEquipment_001();
+
+        ORGANIZATIONID_Type organizationid_type = new ORGANIZATIONID_Type();
+        organizationid_type.setORGANIZATIONCODE("*");
+
+        WOID_Type woid_typeParent = new WOID_Type();
+        woid_typeParent.setJOBNUM(parentWorkorderID);
+        woid_typeParent.setORGANIZATIONID(organizationid_type);
+
+        WOID_Type woid_typeEquipment = new WOID_Type();
+        woid_typeEquipment.setJOBNUM(equipmentID);
+        woid_typeEquipment.setORGANIZATIONID(organizationid_type);
+
+        mp7396_removeWorkOrderEquipment_001.setWORKORDERID(woid_typeParent);
+        mp7396_removeWorkOrderEquipment_001.setRELATEDWORKORDERID(woid_typeEquipment);
+
+        tools.performInforOperation(context, inforws::removeWorkOrderEquipmentOp, mp7396_removeWorkOrderEquipment_001);
+
+        return "OK";
+    }
+
+    /**
+     * Returns a list of all the equipments of the target workorder.
+     *
+     * @param context     the user credentials
+     * @param workorderID the ID of the parent workorder
+     * @return            the list of equipments
+     * @throws InforException
+     */
+    @Override
+    public GridRequestResult getWorkOrderEquipmentsOfWorkorder(InforContext context, String workorderID) throws InforException {
```

Comment:
This should return a list of MECs and not a GridRequestResult

Reply:
(Returning a list of MEC IDs, as discussed)